azkaban-common/src/main/java/azkaban/user/XmlUserManager.java

```diff
@@ -73,6 +80,54 @@ public XmlUserManager(final Props props) {
     this.xmlPath = props.getString(XML_FILE_PARAM);
 
     parseXMLFile();
+
+    // Create a thread which listens to any change in user config file and
+    // reloads it.
+    Runnable runnable = () -> {
+      WatchService watchService;
+      Path path;
+      try {
+        watchService = FileSystems.getDefault().newWatchService();
+        final File file = new File(this.xmlPath);
+        final String dirPath = file.getParent();
+        path = Paths.get(dirPath);
+        path.register(watchService, StandardWatchEventKinds.ENTRY_MODIFY,
+            StandardWatchEventKinds.ENTRY_CREATE);
+      } catch (IOException e) {
+        // Ignore the IOException
+        logger.warn("IOException while setting up watch on conf"
```

Comment:
Is there any reason not to alert administrators right away in a way that is harder to miss?

Reply:
This exception occurs when a new watch service cannot be created or the directory could not be registered with the watch service. If such a thing occurs, the thread would log the warning and return silently. The idea is to keep the system up and running even if this new logic fails to start and avoid any regression.

Reply:
One option would be to send an email to a configurable list of email addresses. The Azkaban email code should be refactored, but that may be out of the scope of this change.